server/sonar-ce/src/test/java/org/sonar/ce/queue/InternalCeQueueImplTest.java

```diff
@@ -530,17 +531,62 @@ public void cancelWornOuts_cancels_pending_tasks_with_executionCount_greater_or_
 
     underTest.cancelWornOuts();
 
-    verifyUnmodifiedByCancelWornOuts(u1);
-    verifyUnmodifiedByCancelWornOuts(u2);
+    verifyUnmodified(u1);
+    verifyUnmodified(u2);
     verifyCanceled(u3);
     verifyCanceled(u4);
-    verifyUnmodifiedByCancelWornOuts(u5);
-    verifyUnmodifiedByCancelWornOuts(u6);
-    verifyUnmodifiedByCancelWornOuts(u7);
-    verifyUnmodifiedByCancelWornOuts(u8);
+    verifyUnmodified(u5);
+    verifyUnmodified(u6);
+    verifyUnmodified(u7);
+    verifyUnmodified(u8);
   }
 
-  private void verifyUnmodifiedByCancelWornOuts(CeQueueDto original) {
+  @Test
+  public void resetTasksWithUnknownWorkerUUIDs_reset_only_in_progress_tasks() {
```

Comment:
I believe it is worth testing the behavior of `resetTasksWithUnknownWorkerUUIDs` in the following cases as well:

1. when an empty set is provided
2. when none of the specified UUIDs has any task in DB (which ever the status) => all pending tasks in DB are reset